src/librustdoc/doctree.rs

```diff
@@ -64,30 +64,51 @@ impl Module<'hir> {
     }
 
     pub(crate) fn push_item(&mut self, new_item: Item<'hir>) {
-        if let Some(existing_item) =
-            self.items.iter_mut().find(|item| item.name() == new_item.name())
-        {
-            if existing_item.from_glob {
-                debug!("push_item: {:?} shadowed by {:?}", *existing_item, new_item);
-                *existing_item = new_item;
-                return;
-            } else if new_item.from_glob {
-                return;
+        if !new_item.name().is_empty() {
+            // todo: also check namespace
```

Comment:
You can use `tcx.def_kind(item.def_id).ns()`. https://doc.rust-lang.org/beta/nightly-rustc/rustc_hir/def/enum.DefKind.html#method.ns